src/main/java/com/truelayer/java/mandates/MandatesHandler.java

```diff
@@ -0,0 +1,55 @@
+package com.truelayer.java.mandates;
+
+import com.truelayer.java.http.entities.ApiResponse;
+import com.truelayer.java.mandates.entities.CreateMandateRequest;
+import com.truelayer.java.mandates.entities.CreateMandateResponse;
+import com.truelayer.java.mandates.entities.ListMandatesQuery;
+import com.truelayer.java.mandates.entities.ListMandatesResponse;
+import com.truelayer.java.mandates.entities.mandatedetail.MandateDetail;
+import com.truelayer.java.payments.entities.AuthorizationFlowResponse;
+import com.truelayer.java.payments.entities.StartAuthorizationFlowRequest;
+import com.truelayer.java.payments.entities.SubmitProviderSelectionRequest;
+import java.util.concurrent.CompletableFuture;
+import lombok.Value;
+
+@Value
+public class MandatesHandler implements IMandatesHandler {
+    IMandatesApi mandatesApi;
+
+    @Override
+    public CompletableFuture<ApiResponse<CreateMandateResponse>> createMandate(CreateMandateRequest request) {
+        return mandatesApi.createMandate(request);
+    }
+
+    @Override
+    public CompletableFuture<ApiResponse<AuthorizationFlowResponse>> startAuthorizationFlow(
+            String mandateId, StartAuthorizationFlowRequest request) {
+        return mandatesApi.startAuthorizationFlow(mandateId, request);
+    }
+
+    @Override
+    public CompletableFuture<ApiResponse<AuthorizationFlowResponse>> submitProviderSelection(
+            String mandateId, SubmitProviderSelectionRequest request) {
+        return mandatesApi.submitProviderSelection(mandateId, request);
+    }
+
+    @Override
+    public CompletableFuture<ApiResponse<ListMandatesResponse>> listMandates() {
+        return mandatesApi.listMandates(null, null, null);
```

Comment:
Is this returning all mandates or do we have a default limit (if not specified) on PGW which will enforce pagination?